src/util/inconsistencies.py

```diff
@@ -24,6 +23,99 @@
 FROM_ENV = object()
 
 
+def break_recursive_loops(cr, model, field, name_field="name"):
+    # TODO add a variant to verify loops on m2m
+    _validate_model(model)
+
+    table = table_of_model(cr, model)
+    trgt = target_of(cr, table, field)
+    if not trgt or trgt[:2] != (table, "id"):
+        raise ValueError("The column `{}` is not FK on itself".format(field))
+
+    query = format_query(
+        cr,
+        """
+        WITH RECURSIVE __loop AS (
+         SELECT array[{field}] AS path,
+                False AS cycle
+           FROM {table}
+          WHERE {field} IS NOT NULL
+       GROUP BY {field}
+          UNION ALL
+         SELECT child.{field} || curr.path AS path,
+                child.{field} = any(curr.path) AS cycle
+           FROM __loop AS curr
+           JOIN {table} AS child
+             ON child.id = curr.path[1]
+          WHERE child.{field} IS NOT NULL
+            AND NOT curr.cycle
```

Comment:
Simpler?
```suggestion
         SELECT child.id || curr.path AS path,
                child.id = any(curr.path) AS cycle
           FROM __loop AS curr
           JOIN {table} AS child
             ON child.{field} = curr.path[1]
```

Reply:
The `AND NOT curr.cycle` is needed to actually stop the loop (because of the `UNION ALL`)

Same for the `child.{field} IS NOT NULL`, it skips the last hop in the tree.

Reply:
Yep, I removed too much in the where...

Reply:
> Same for the child.{field} IS NOT NULL, it skips the last hop in the tree.

Since path has non-null values, `child.{field} = path[1]` should avoid the need to check non null field... but anyway it doesn't add much...

Reply:
I think your version is fine. You pick all parents with at least one child to start, then in the recursive step you take all parents of any record already in the table. Seems correct. My suggestion was in the other direction: to add children of what we already have. Both are equivalent I think.

Reply:
ofc the removing the check for cycle was wrong in my suggestion